src/client/testing/common/xUnitParser.ts

```diff
@@ -49,10 +51,27 @@ export class XUnitParser implements IXUnitParser {
     constructor(@inject(IFileSystem) private readonly fs: IFileSystem) {}
 
     // Update "tests" with the results parsed from the given file.
-    public async updateResultsFromXmlLogFile(tests: Tests, outputXmlFile: string) {
-        const data = await this.fs.readFile(outputXmlFile);
+    public async updateResultsFromXmlLogFile(tests: Tests, outputXmlFile: string, timeout = 0) {
+        var data = '';
+        var parserResult = undefined;
+        const timer = new StopWatch();
+        // Poll for valid xml
+        do {
+            data = await this.fs.readFile(outputXmlFile);
+            if (data) {
+                try {
+                    parserResult = await parseXML(data);
+                } catch (error) {}
+                if (parserResult) {
+                    break;
+                }
+            }
+            await sleep(100);
+        } while (timer.elapsedTime < timeout);
```

Comment:
The linter will probably complain about the `break`.  Here's an adjustment that should make it happy:

```suggestion
        async function readResult() {
            const data = await this.fs.readFile(outputXmlFile);
            if (!data) {
                return undefined;
            }
            try {
                return parseXML(data);
            } catch {
                return undefined;
            }
        }
        let parserResult = await readResult();
        while (!parserResult && timer.elapsedTime < timeout) {
            await sleep(100);
            parserResult = await readResult();
        }
```